src/sources/socket/tcp.rs

```diff
@@ -7,19 +7,29 @@ use crate::{
 };
 use bytes::Bytes;
 use codec::BytesDelimitedCodec;
+use getset::{CopyGetters, Getters, Setters};
 use serde::{Deserialize, Serialize};
 
-#[derive(Deserialize, Serialize, Debug, Clone)]
-#[serde(deny_unknown_fields)]
+#[derive(Deserialize, Serialize, Debug, Clone, Getters, CopyGetters, Setters)]
```

Comment:
I'd rather we don't introduce the getter/setter pattern for just this one component unless there it solves a need for you, or there is agreement this is a direction we want to take the Vector code base. (Yes I see we use it for the log schema, but only that component AFAICT.)

Reply:
Hm, I wanted to get rid of the `pub` fields here as we have in other places. Do you have a suggestion what to do here instead?